toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,368 @@
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device = findBootIfValue()
+	}
+
+	for _, iface := range ifaces {
+		if networkData.Device == iface.Name {
+			deviceName = networkData.Device
+			return
+		} else {
+			ifaceAddr := strings.TrimSpace(iface.HardwareAddr.String())
+
+			if strings.Contains(networkData.Device, ":") {
+				networkData.Device = strings.TrimSpace(networkData.Device)
+
+				if ifaceAddr != "" && strings.EqualFold(ifaceAddr, networkData.Device) {
+					deviceName = iface.Name
+					return
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	const (
+		id        = "[Match]\n"
+		NameField = "Name="
+	)
+
+	matchSection := id + NameField + deviceName + "\n"
+	err = file.Append(matchSection+"\n", fileName)
+	if err != nil {
+		logger.Log.Infof("Failed to write [Match] section: %s", err)
+	}
+
+	return
+}
+
+func populateNetworkSection(networkData Network, fileName string) (err error) {
+	const (
+		id           = "[Network]\n"
+		ipField      = "Address="
+		gateWayField = "Gateway="
+		dnsField     = "DNS="
+	)
+
+	networkSection := id
+
+	// Currently only supports static IP setting
+	// Update IP and netmask
+	netMask := strings.Trim(networkData.NetMask, " ")
+	ip := strings.Trim(networkData.NetMask, " ")
+
+	if netMask != "" && ip != "" {
+		stringMask := net.IPMask(net.ParseIP(networkData.NetMask).To4())
+		cidrPrefix, _ := stringMask.Size()
+		ipAddr := networkData.Ip + "/" + strconv.Itoa(cidrPrefix)
+
+		networkSection = networkSection + ipField + ipAddr + "\n"
+	}
+
+	// Update Gateway
+	gateway := strings.Trim(networkData.GateWay, " ")
+	if gateway != "" {
+		networkSection = networkSection + gateWayField + networkData.GateWay + "\n"
+	}
+
+	// Update Nameserver
+	for _, nameserver := range networkData.NameServer {
+		dnsSetting := dnsField + nameserver + "\n"
+		networkSection = networkSection + dnsSetting
+	}
+
+	err = file.Append(networkSection, fileName)
+	if err != nil {
+		logger.Log.Infof("Failed to write [Network] section: %s", err)
+	}
+
+	return
+}
+
+func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Network, deviceName string) (err error) {
+	const (
+		networkFileDir = "/etc/systemd/network/"
+		filePrefix     = "10-"
+	)
+
+	if exists, ferr := file.DirExists(networkFileDir); ferr != nil {
+		logger.Log.Errorf("Error accessing /etc/systemd/network/")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("/etc/systemd/network/: no such path or directory")
+		return
+	}
+
+	networkFileName := networkFileDir + filePrefix + networkData.BootProto + "-" + deviceName + ".network"
+	if exists, _ := file.PathExists(networkFileName); exists {
+		return fmt.Errorf("Network file (%s) already exists", networkFileName)
+	}
+
+	err = file.Create(networkFileName, 0644)
+	if err != nil {
+		logger.Log.Errorf("Error creating file %s: %s", networkFileName, err)
```

Comment:
In two places you're using `%w` for printing `err`. Let's be consistent and unify these formats.